### PR TITLE
👽️(API) :  Initial API

### LINKED ISSUE #28: Connect mongo to spring application
Create interfaces in spring app for mongo connection

### LINKED ISSUE #46: Create Controllers
Create Controllers for all models in Spring

### LINKED ISSUE #47: Create API service for Angular
Create service for angular to interact with backend

### PR DESCRIPTION
**Pull request by priyo5**
Initial structure and first use case functionality 
- Added Controllers for all collections
- Added API service for Angular
Closes #28 
Closes #47 
Closes #46

### CODE REVIEW

File: Server/src/main/java/Database/TheAuPair/Controllers/ActivityController.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package Database.TheAuPair.Controllers;
+
+import Database.TheAuPair.Models.Activity;
+import Database.TheAuPair.Repositories.ActivityRepository;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+
```
Comment (Ruben-Brits) [inline Server/src/main/java/Database/TheAuPair/Controllers/ActivityController.java:7]: empty line

File: libs/shared/interfaces/activity.interfaces.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    name: string;
+    description: string;
+    location: string;
+    timeStart: string;
+    timeEnd: string;
+    budget: number;
+    comment: string;
+    behavior: string;
+    day: string ;
+  }
```
Comment (PxttytheThxtty) [inline libs/shared/interfaces/activity.interfaces.ts:12]: Im sorry

File: Server/src/main/java/Database/TheAuPair/Repositories/ActivityRepository.java [hunk 3]
```diff
@@ -2,8 +2,14 @@
 
 import Database.TheAuPair.Models.Activity;
 import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.stereotype.Repository;
 import org.springframework.data.mongodb.repository.Query;
```
Comment (Ruben-Brits) [inline Server/src/main/java/Database/TheAuPair/Repositories/ActivityRepository.java:7]: empty line

File: Server/src/main/java/Database/TheAuPair/Repositories/auPairRepository.java [hunk 4]
```diff
@@ -2,8 +2,14 @@
 
 import Database.TheAuPair.Models.auPair;
 import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.stereotype.Repository;
 import org.springframework.data.mongodb.repository.Query;
```
Comment (Ruben-Brits) [inline Server/src/main/java/Database/TheAuPair/Repositories/auPairRepository.java:7]: empty line

File: libs/shared/interfaces/activity.interfaces.ts (hunk 2 above)
Comment (Ruben-Brits) [inline libs/shared/interfaces/activity.interfaces.ts:12]: newline

**Review by Ruben-Brits — APPROVED** [review]

**Review by d-azmanov — APPROVED** [review]

**Review by PxttytheThxtty — APPROVED** [review]

**Pull request merged**